### DISCUSSION TITLE
What's the difference between a green "Velocity" point and a purple "Scroll Speed" timing point?

### DISCUSSION THREAD

**Discussion by zachmanthethird** [ppy/osu#15808] [Q&A] [answered]
Continuation of #15747 with hopefully some more clarity.

Okay, here's my understanding, and please feel free to correct me if I am wrong:

Green points labelled "Velocity" are available in all modes. These points are locked between 0.1 and 10. These correspond to stable's Slider Velocity and represent the speed at which the slider travels. By keeping distance constant, they directly influence the time a slider takes to complete; therefore, by changing the "Velocity" value, you must also change the slider distance to keep the end snapping.

Purple timing points labelled "Scroll Speed" are only available in taiko and mania; other modes do not save these points. These points are extended from 0.01 to 10. This is _meant_ to mirror how Slider Velocity acts in these two modes in stable by changing the scroll speed of the playfield. Therefore, changing these values should not affect sliders' (or drumrolls and hold notes, respectively) distance or timing snappings.

But these points are serving the same purpose?? In stable, these are combined into a single value: Slider Velocity. Additionally, there are a few flaws present by splitting them:

1. By not allowing purple points in std or catch, green and purple points must have the same limits. Otherwise, you run into problems such as https://osu.ppy.sh/b/1494828 where green points can't reach 0.01, causing the converted map in taiko or mania to be unplayable.
2. Because the stable editor was designed only for std, lazer attempts to convert these to taiko maps with minimal success. From what I can tell, sliders are converted with green timing points, and anything else is converted to purple timing points. However, if there are notes after a slider without another SV point, a green and purple point are created simultaneously. Green points are editable; however, when saving, the purple point overwrites the green value causing unexpected errors. Because the green point changes the length of a slider, you can create infinitely long sliders simply by changing the green value and not the purple value. This is what caused my confusion when editing https://osu.ppy.sh/b/2294946.

Unfortunately, I cannot provide any solutions, but I hope that this helps identify the problems that may lead to solutions.

**Comment by peppy** [comment, marked as answer]
slider velocity is not a global concept in the new editor so i don't know what you mean here. it has been moved to per-object and implemented only in rulesets that have sliders.

scroll speed affects the global playfield scroll speed.

**Reply by zachmanthethird** [reply]
Ahh, okay, so green points are per-object, not global. This makes more sense. So this should be rewritten:

2. Sliders are converted with green SV points, and these are editable in the same way as in std and catch. However, upon saving, their values are reset to the global purple "Scroll Speed" values upon saving, and slider lengths are not reduced accordingly.

Point 1 still stands valid as-is.

**Reply by peppy** [reply]
I'm not sure where this is written. Also please don't refer to these things by colour. We never use colours to discern these mechanics and the colours may change at any point.

I'm not sure what you mean by "reset". Could you provide a video showing what you mean? It sounds like a bug if so. Nothing should be reset.

**Reply by zachmanthethird** [reply]
Here's a video using https://osu.ppy.sh/b/2294946:
https://user-images.githubusercontent.com/74262407/143511377-a35f25a4-5a37-4417-b315-b01faa390b08.mp4
Additionally, this error log was thrown one of the times when replicating the issue:
[runtime.log.txt](https://github.com/ppy/osu/files/7605641/runtime.log.txt)

**Reply by peppy** [reply]
the timeline adjustment should probably not be there on taiko, that is an oversight
